modules/logging/filewriter.go

```diff
@@ -47,7 +48,11 @@ func (FileWriter) CaddyModule() caddy.ModuleInfo {
 }
 
 func (fw FileWriter) String() string {
-	fpath, err := filepath.Abs(fw.Filename)
+	// Replace placeholder in filename
+	repl := caddy.NewReplacer()
+	filename := repl.ReplaceKnown(fw.Filename, "")
```

Comment:
This is a good idea, to support placeholders in log file names.

But we should do the replacement only once.

All Caddy modules go through a provisioning phase before they are used: https://github.com/caddyserver/caddy/wiki/v2:-Writing-a-Module#provisioning

So, I would:

1) make a `Provision()` method that implements the `caddy.Provisioner` interface,
2) set an [interface guard](https://github.com/caddyserver/caddy/wiki/v2:-Writing-a-Module#interface-guards) - (see examples in other files, you can do a search in the code for "interface guard")
3) in the `Provision()` method, do the replacement there.

This way, we can use `fw.Filename` just as before, because it will have already been evaluated for replacements during provisioning.